app/assets/javascripts/app/views/comment_stream_view.js

```diff
@@ -109,6 +184,148 @@ app.views.CommentStream = app.views.Base.extend({
+        Diaspora.I18n.t("photo_uploader.error", {file: fileName}) +
+        "</div>"
+      );
+    }
+  },
+
+  // add photo placeholder to the publisher to indicate an upload in progress
+  _addPhotoPlaceholder: function() {
+    var id = this.photozoneEl.find('li.publisher_photo').length;
+
+    this.photozoneEl.append(
+      "<li class=\"publisher_photo loading\" style=\"position:relative;\">" +
+      "  <div class=\"progress\">" +
+      "    <div class=\"progress-bar progress-bar-striped active\" role=\"progressbar\"></div>" +
+      "  </div>" +
+      "  <img data-id=\"" + id +"\" src=\"\"+Handlebars.helpers.imageUrl(\"ajax-loader2.gif\")+\"\"" +
+      "    class=\"ajax-loader\" alt=\"\" />" +
+      "</li>"
+    );
+    this.textareaWrapperEl.addClass("with_attachments");
+  },
+
+  // replace the first photo placeholder with the finished uploaded image and
+  // add the id to the publishers form
+  _addFinishedPhoto: function(id, url) {
+    // add form input element
+    this.$el.append(
+      "<input type=\"hidden\" value=\""+id+"\" name=\"photos[]\" />"
+    );
+
+    // replace placeholder
+    var placeholder = this.photozoneEl.find("li.loading").first();
+        imgEl = placeholder.find("img");
+
+    placeholder
+      .removeClass("loading")
+      .prepend("<div class=\"x\"></div>");
+
+    imgEl.attr({"src": url, "data-id": id}).removeClass("ajax-loader");
+
+    placeholder.find("div.progress").remove();
+
+    this.updateTexareaValue(
+      this.textareaEl.val() + "![" +
+      Diaspora.I18n.t("publisher.markdown_editor.texts.insert_image_description_text") +
+      "](" + imgEl[0].src + " \"" +
+      Diaspora.I18n.t("publisher.markdown_editor.texts.insert_image_title") + "\")"
+    );
+  },
+
+  _cancelPhotoUpload: function() {
+    var placeholder = this.photozoneEl.find("li.loading").first();
+
+    placeholder
+      .removeClass("loading")
+      .find("img").remove();
+  },
+
+  // remove an already uploaded photo
+  _removePhoto: function(evt) {
+    var self  = this;
+    var photo = $(evt.target).parents(".publisher_photo");
+
+    this.removePhoto(photo);
+
+    return false;
+  },
+
+  removePhoto: function(photo) {
+    photo.addClass("dim");
+
+    var img = photo.find("img");
+
+    if (!img.length) {
+      photo.remove();
+      return;
+    }
+
+    var imgUrl = img[0].src,
+        markdownRegex = new RegExp("!\\[[^\\]]*\\]\\(" + imgUrl + "( *|( +\"[^\"]*\" *))?\\)", "g"),
+        self = this;
+
+    $.ajax({
+      url: "/photos/" + img.attr("data-id"),
+      dataType: "json",
+      type: "DELETE",
+      success: function() {
+        $.when(photo.fadeOut(400)).then(function() {
+          photo.remove();
+          self.updateTexareaValue(self.textareaEl.val().replace(markdownRegex, ""));
+
+          if (self.$(".publisher_photo").length === 0) {
+            // no more photos left...
+            self.textareaWrapperEl.removeClass("with_attachments");
+          }
```

Comment:
- Expected indentation of 4 space characters but found 8.
- 'imgEl' is not defined.